src/qt/walletmodel.h

```diff
@@ -262,7 +262,25 @@ class WalletModel : public QObject
     void getOutputs(const std::vector<COutPoint>& vOutpoints, std::vector<COutput>& vOutputs);
     bool getMNCollateralCandidate(COutPoint& outPoint);
     bool isSpent(const COutPoint& outpoint) const;
-    void listCoins(std::map<QString, std::vector<COutput> >& mapCoins) const;
+
+    class ListCoinsKey {
+    public:
+        QString address;
+        Optional<QString> stakerAddres; // used only for P2CS utxo
+
+        bool operator==(const ListCoinsKey& key2) const {
+            if (this->address != key2.address) return false;
+            if ((!this->stakerAddres && key2.stakerAddres) ||
+                    (this->stakerAddres && !key2.stakerAddres)) return false;
+            return *this->stakerAddres == *key2.stakerAddres;
```

Comment:
Boost optionals are equality-comparable if the underlying type is as well (https://www.boost.org/doc/libs/1_59_0/libs/optional/doc/html/boost_optional/tutorial/relational_operators.html), and QString is.  So we can simplify this as
```cpp
return address == key2.address && stakerAddress == key2.stakerAddress;
```

Reply:
done